<?xml version="1.0"?>
<launch>
  <!-- TARGET DRONE CONFIG  -->
  <arg name="bar_l"           value="1.2"/>
  <arg name="ball_r"          value="0.09"/>
  <arg name="ball_colour"     value="Gazebo/green"/>
  <arg name="x"               value="1"/>
  <arg name="y"               value="1"/>
  <arg name="z"               value="0"/>
  <arg name="R"               value="0"/>
  <arg name="P"               value="0"/>
  <arg name="Y"               value="0"/>
  <arg name="speed"           value="5"/>
  <arg name="trajectory_a"    value="40"/>
  <arg name="trajectory_b"    value="4"/>

  <!-- vehicle model and world -->
  <arg name="est"           default="ekf2"/>
  <arg name="vehicle"       default="iris"/>

  <!-- Challenge 1 scenario paramaters -->
  <arg name="balloon_zone_x_size"   value="80"/>
  <arg name="balloon_zone_y_size"   value="40"/>
  <arg name="max_balloons_height"   value="4"/>
  <arg name="csv_path"              value="$(find challenge_worlds)/config/dist1.csv"/>
  <arg name="deploy_zone_size"      value="2"/>
  <arg name="landing_zone_size"     value="10"/>
  <arg name="deploy_zone_x"         value="43"/>
  <arg name="deploy_zone_y"         value="6"/>
  <arg name="landing_zone_x"        value="50"/>
  <arg name="landing_zone_y"        value="6"/>

  <!-- gazebo configs -->
  <arg name="gui"            default="true"/>
  <arg name="debug"          default="false"/>
  <arg name="verbose"        default="true"/>
  <arg name="paused"         default="true"/>
  <arg name="respawn_gazebo" default="false"/>

  <!-- MAVROS configs -->
  <arg name="fcu_url"        default="udp://:14540@localhost:14557"/>
  <arg name="respawn_mavros" default="false"/>
  <arg name="gcs_url"        default="" />
	<arg name="tgt_system"     default="1" />
	<arg name="tgt_component"  default="1" />
	<arg name="log_output"     default="screen" />
	<arg name="fcu_protocol"   default="v2.0" />

  <!-- PX4 configs -->
  <arg name="interactive"    default="true"/>
  <arg name="ID" default="0"/>
  <env name="PX4_SIM_MODEL" value="$(arg vehicle)" />
  <env name="PX4_ESTIMATOR" value="$(arg est)" />
  <arg unless="$(arg interactive)" name="px4_command_arg1" value=""/>
  <arg     if="$(arg interactive)" name="px4_command_arg1" value="-d"/>


	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
		<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)" />
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
		<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
		<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
	</include>

  <node name="sitl_$(arg ID)" pkg="px4" type="px4" output="screen"
    args="$(find px4)/ROMFS/px4fmu_common -s etc/init.d-posix/rcS -i $(arg ID) $(arg px4_command_arg1)" >
  </node>

  <include file="$(find challenge_worlds)/launch/arena.launch">
    <arg name="balloon_zone_x_size"   value="$( arg balloon_zone_x_size)"/>
    <arg name="balloon_zone_y_size"   value="$( arg balloon_zone_y_size)"/>
    <arg name="max_balloons_height"   value="$( arg max_balloons_height)"/>
    <arg name="csv_path"              value="$( arg csv_path)"/>
    <arg name="deploy_zone_size"      value="$( arg deploy_zone_size)"/>
    <arg name="landing_zone_size"     value="$( arg landing_zone_size)"/>
    <arg name="deploy_zone_x"         value="$( arg deploy_zone_x)"/>
    <arg name="deploy_zone_y"         value="$( arg deploy_zone_y)"/>
    <arg name="landing_zone_x"        value="$( arg landing_zone_x)"/>
    <arg name="landing_zone_y"        value="$( arg landing_zone_y)"/>
  </include>


 <param command="$(find xacro)/xacro -i
   $(find challenge_worlds)/urdf/iris_target_drone.xacro
   bar_l:=$(arg bar_l) ball_r:=$(arg ball_r) ball_colour:=$(arg ball_colour)"
   name="target_iris_drone" />

 <param command="$(find xacro)/xacro -i
    $(find challenge_worlds)/urdf/iris_support.xacro
    bar_l:=$(arg bar_l) ball_r:=$(arg ball_r) "
    name="target_iris_support" />

 <node name="support_spawn" pkg="gazebo_ros" type="spawn_model"
   output="screen" args="-sdf -param target_iris_support -model support
   -x $(arg x) -y $(arg y) -z $(arg z) "/>

 <node name="$(anon vehicle_spawn)" pkg="gazebo_ros" type="spawn_model"
   output="screen" args="-sdf -param target_iris_drone -model $(arg vehicle)
   -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y) "/>

</launch>
